src/BlazorUI/Bit.BlazorUI/Components/Inputs/SearchBox/BitSearchBox.razor.cs

```diff
@@ -261,6 +263,13 @@ protected override Task OnInitializedAsync()
         return base.OnInitializedAsync();
     }
 
+    protected override void OnParametersSet()
+    {
+        _searchButtonIconName = Dir is BitDir.Rtl ? "ChromeBack" : "ChromeBackMirrored";
```

Comment:
please refactor this approach as we've discussed.

Reply:
We didn't fully agree with an "approach" in our last discussion! :)